### PR TITLE
Feature/ ready-jobs and server stats publication

### PR DESCRIPTION
**Pull request by rhyslbw**
This PR includes a refactor that I felt was necessary to support the features. It creates a more reasonable and self-contained set of objects that make up the module IMO.
- Job Collection is now configured within and controlled by a singleton, rather than in the module. This is inline with the Event Sourcing [Snapshoter](https://github.com/meteor-space/event-sourcing/blob/develop/source/server/infrastructure/snapshotter.coffee) collection, and a good pattern to keep the module clear of this clutter.
- An in-memory collection indicates the current running state of the jobServer, useful for showing this in the UI by subscribing the the publication.
- A persistant collection contains a record of all subscribed external workers, with the logic to clear out orphaned records when the server terminates without the subscriptions being stopped. As part of the subscription, the client provides the previous session id that is used to check for this exception.
- [Promote](https://github.com/vsivsi/meteor-job-collection#jcpromotemilliseconds---server-only) interval is also now configurable, which just uses the default JobCollection value. This will be important for a system with multiple jobServer instances as the promote then becomes adjusted to ensure there are no race conditions.
- Logger can be controlled at the module level now.

### CODE REVIEW

File: source/server/publications.js [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+          // Logged in users only -> later will be capability-based
+          if(context.userId === undefined) {
+            throw new Meteor.Error('Unauthorised Access');
+          }
+          return this.connectedWorkers.find();
+        }
+      });
+    }
+    Space.messaging.Publication.prototype.onDependenciesReady.call(this);
+  },
```
Comment (DominikGuzei) [inline source/server/publications.js]: Why not move all contents of the `onDependenciesReady` method into `publications` and return the array at the end?

Comment (rhyslbw) [inline source/server/publications.js]: Yer… haha. Done

**Comment by rhyslbw** [comment]
@darko-mijic can you also review and comment as would be good to see this released as 0.7.0 for you to start using it without mgp. Sorry for the lack of scope here @DominikGuzei, the refactor should have been separated out into a new PR

**Comment by DominikGuzei** [comment]
@rhyslbw and @darko-mijic i just trust you with this package, since i have no real-world use case for it now, it's a bit hard to have an opinion about the structure etc. :wink:

**Comment by rhyslbw** [comment]
Great, just the once over is all i'm looking for, regarding code quality. happy to proceed now as it will be pre 1.0

**Pull request merged**